File: LinkedList.java
import java.util.Scanner;
class LinkedListNode{
	int data;
	LinkedListNode next;
}
public class LinkedList{
	LinkedListNode root;
	LinkedListNode prev;
	public void insert(int data) {
		LinkedListNode node  = new LinkedListNode();
		node.data = data;
		node.next = null;
		if(root == null) {
			root  = node;
			prev = node;
		}
		else {
			prev.next = node;
			prev = node;
		}
	}
	public void swapNode(int x, int y) {
		if (x == y) return; 
        LinkedListNode a = null, temp1 = root; 
        while (temp1 != null && temp1.data != x) 
        { 
            a = temp1; 
            temp1 = temp1.next; 
        }
        LinkedListNode b = null, temp2 = root; 
        while (temp2 != null && temp2.data != y) 
        { 
            b = temp2;
            temp2 = temp2.next; 
        } 
        if (temp1 == null || temp2 == null) 
            return; 
        if (temp1 != null) 
            a.next = temp2; 
        else 
            root = temp2; 
        if (temp2 != null) 
            b.next = temp1; 
        else 
            root = temp1; 
        LinkedListNode temp = temp1.next; 
        temp1.next = temp2.next; 
        temp2.next = temp; 
	}
	public void disp() {
		LinkedListNode cur = root;
		while(cur!=null) {
			System.out.println(cur.data);
			cur = cur.next;
		}
	}
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in); 
		int n = sc.nextInt();
		LinkedList linkedList  = new LinkedList();
		for(int i =0;i<n;i++) {
			int data  = sc.nextInt();
			linkedList.insert(data);
		}
		linkedList.disp();
		linkedList.swapNode(2,4);
		linkedList.disp();
		sc.close();
	}
}

File: linkedlist.py
class Node:

	def __init__(self, data):
		self.data = data
		self.next = None


class LinkedList:

	def __init__(self):
		self.head = None
		self.temp = self.head

	def insert(self, data):
		if self.head == None:
			self.head = Node(data)
			self.temp = self.head;
		else:
			node = Node(data)
			self.temp.next = node
			self.temp = node
	
	def insertAtPosition(self, data, pos):
		temp = self.head
		prev = None
		for i in range(0,pos):
			prev = temp
			temp = temp.next
			if temp.next == None:
				break
		node = Node(data)
		if pos <= 0:
			node.next = temp
			self.head = node
		elif temp.next == None:
			temp.next = node
		else:
			node.next = temp
			prev.next = node

	def reverse(self):
		prev = None
		cur = self.head;
		while cur!=None:
			temp = cur.next; 
			cur.next = prev;
			prev = cur
			cur = temp
		self.head = prev


	def delete(self, data):
		temp = self.head
		prev = None
		cur = None		
		if temp.data == data:
			self.head = temp.next
			temp.next == None
		elif temp.next == None:
			self.head = None
			return
		else:
			while(temp!=None):
				if temp.data == data:
					cur = temp.next
					temp.next = None
					temp = None
					break
				else:
					prev = temp
				temp = temp.next
			if cur == None:
				prev.next = None
			else:
				prev.next = cur

	def disp(self):
		self.temp = self.head
		if self.head == None:
			return
		while(self.temp!=None):
			print(self.temp.data)
			self.temp = self.temp.next

if __name__ == '__main__':
	linkedList = LinkedList()
	n  = int(input())
	for i in range(0,n):
		data = int(input())
		linkedList.insert(data)
	pos = int(input())
	data = int(input())
	linkedList.insertAtPosition(data, pos)
	linkedList.disp()
	deletedata = int(input())
	linkedList.delete(deletedata)
	print("after delete")
	linkedList.reverse()
	linkedList.disp()